audb/core/load.py

```diff
@@ -1037,6 +1037,116 @@ def load(
     return db
 
 
+def load_attachment(
+        name: str,
+        attachment: str,
+        *,
+        version: str = None,
+        cache_root: str = None,
+        num_workers: typing.Optional[int] = 1,
+        verbose: bool = True,
+) -> audformat.Database:
+    r"""Load attachment(s) of database.
+
+    Args:
+        name: name of database
+        attachments: attachment ID to load
+        version: version of database
+        cache_root: cache folder where databases are stored.
+            If not set :meth:`audb.default_cache_root` is used
+        num_workers: number of parallel jobs or 1 for sequential
+            processing. If ``None`` will be set to the number of
+            processors on the machine multiplied by 5
+        verbose: show debug messages
+
+    Returns:
+        list of attachment file paths
+
+    Raises:
+        ValueError: if an attachment ID is requested
+            that is not part of the database
+
+    Examples:
+        >>> paths = load_attachment(
+        ...     'emodb',
+        ...     'bibtex',
+        ...     version='1.4.1',
+        ...     verbose=False,
+        ... )
+        >>> os.path.basename(paths[0])
+        'burkhardt2005emodb.bib'
+
+    """
+    if version is None:
+        version = latest_version(name)
+
+    db_root = database_cache_root(name, version, cache_root)
+
+    if verbose:  # pragma: no cover
+        print(f'Get:   {name} v{version}')
+        print(f'Cache: {db_root}')
+
+    deps = dependencies(
+        name,
+        version=version,
+        cache_root=cache_root,
+        verbose=verbose,
+    )
+    # We use single archive per attachment ID,
+    # so we can infer available attachment IDs
+    # from archive names
+    attachment_ids = list(
+        deps._df[
+            deps._df['type'] == define.DependType.ATTACHMENT
+        ].archive
+    )
+    if attachment not in attachment_ids:
+        msg = error_message_missing_object(
+            'attachment',
+            [attachment],
+            name,
+            version,
+        )
+        raise ValueError(msg)
+    attachment_files = list(
+        deps._df[
+            deps._df['archive'] == attachment
+        ].index
+    )
+
+    with FolderLock(db_root):
+
+        # Start with database header
+        db, backend = load_header_to(
+            db_root,
+            name,
+            version,
+        )
+
+        # Load attachments
+        _load_files(
```

Comment:
I wonder if it was in fact a good idea to introduce this generic function in https://github.com/audeering/audb/pull/244 instead of a separate `_load_attachment()`. What it does is: 1. figure out what files are in an attachment, 2. figure out which files are missing 3. figure out to which archive the missing files belong. But since we store all files of an attachment in a single archive this doesn't make sense since either all files of an attachment are missing or none. And also the files are never distributed over multiple archives. So I think we can speed this function up a lot by checking if the attachment path exists, then we are done or if it does not exist we download the archive and unpack it. 


Because we now each attachment is stored in a single archive it doesn't really make sense to   What it does is: it first figure out what files are in an attachment and then

Reply:
But if we decided to change this we should probably do this in a separate PR.

Reply:
Yes, please open another issue for this. If we really want to go back to have different code for media, table, attachment, I would also vote for having extra `audb/core/load_table.py`, `audb/core/load_media.py`, `audb/core/load_attachment.py` files that handle the code for the particular media type.

Reply:
I created https://github.com/audeering/audb/issues/275